spinalcordtoolbox/deepseg/models.py

```diff
@@ -0,0 +1,104 @@
+# coding: utf-8
+"""
+Deals with models for deepseg module. Available models are listed under MODELS.
+"""
+
+
+import os
+import json
+import logging
+
+import spinalcordtoolbox as sct
+import spinalcordtoolbox.download
+
+
+logger = logging.getLogger(__name__)
+
+# List of models. The convention for model names is: (species)_(university)_(contrast)_region
+# Regions could be: sc, gm, lesion, tumor
+# TODO: add mirror
+# TODO: add a test to make sure all fields are present in the dict below
+# TODO: deal with "Folder __MACOSX"
+MODELS = {
+    't2star_sc':
+        {'url': 'https://osf.io/v9hs8/download?version=2',
+         'description': 'Cord segmentation on T2*-weighted contrast.',
+         'default': True},
+    'mice_uqueensland_sc':
+        {'url': 'https://osf.io/nu3ma/download?version=4',
+         'description': 'Cord segmentation on mouse MRI. Data from University of Queensland.',
+         'default': False},
+    'mice_uqueensland_gm':
+        {'url': 'https://osf.io/mfxwg/download?version=4',
+         'description': 'Gray matter segmentation on mouse MRI. Data from University of Queensland.',
+         'default': False},
+    }
+
+
+def folder(name_model):
+    """
+    Return absolute path of deep learning models.
+    :param name: str: Name of model.
+    :return: str: Folder to model
+    """
+    return os.path.join(sct.__models_dir__, name_model)
+
+
+def install_model(name_model):
+    """
+    Download and install specified model under SCT installation dir.
+    :param name: str: Name of model.
+    :return: None
+    """
+    logger.info("\nINSTALLING MODEL: {}".format(name_model))
+    sct.download.install_data(MODELS[name_model]['url'], os.path.split(folder(name_model))[0])
+
+
+def install_default_models():
+    """
+    Download all default models and install them under SCT installation dir.
+    :return: None
+    """
+    for name_model, value in MODELS.items():
+        if value['default']:
+            install_model(name_model)
+
+
+def is_installed(name_model):
+    """
+    Check if model is installed under SCT directory.
+    :param name: str: Name of model.
+    """
+    if os.path.exists(os.path.join(folder(name_model), name_model + '.pt')) and \
+            os.path.exists(os.path.join(folder(name_model), name_model + '.json')):
+        return True
+    else:
+        return False
+
+
+def list_models():
```

Comment:
This is UI, so it should be in scripts/sct_deepseg.py.

I would split the listing from the displaying: keep, in this file, this version:

```
def list_models():
    # ...
    return {name: value for name, value in MODELS.items() if is_installed(name)}
```

and move the parts that fiddle with terminal colours over to `scripts/sct_deepseg.py`.

Also, at the expense of an extra dependency, you can make colouration more readable with https://pypi.org/project/colored/.

Reply:
@kousu i've done your suggestion, and added dependency with colored in a2b8abe0c5edea44cca969798e1c43f4fbda2eb3, but i'm getting crazy with a formatting that doesn't seem to work when using `colored`-- i'm not sure if it's a bug and i don't have the time to dig. It is producing this:
~~~
MODEL                    DESCRIPTION
--------------------------------------------------------------------------------
t2star_sc[*]Cord segmentation on T2*-weighted contrast.
~~~
Instead of that:
~~~
MODEL                    DESCRIPTION
--------------------------------------------------------------------------------
t2star_sc               [*]Cord segmentation on T2*-weighted contrast.
~~~

Reply:
It looks okay over here

```
$ sct_deepseg -list-models

--
Spinal Cord Toolbox (git-jca/2626-deepseg-0f537cb59edb3c780f718d28304be45da11d6fe0)

MODEL                    DESCRIPTION
--------------------------------------------------------------------------------
t2star_sc[*]             Cord segmentation on T2*-weighted contrast.
mice_uqueensland_sc      Cord segmentation on mouse MRI. Data from University of Queensland.
mice_uqueensland_gm      Gray matter segmentation on mouse MRI. Data from University of Queensland.

Legend: installed | not installed | default: [*]
```

![screenshot -- with colours](https://user-images.githubusercontent.com/987487/82130166-e3259100-9796-11ea-84bf-f0250c3e9584.png)

Reply:
yup-- sorry i should have mentioned this is now fixed